modules/relayers/src/lib.rs

```diff
@@ -70,9 +71,28 @@ pub mod pallet {
 		/// Stake and slash scheme.
 		type StakeAndSlash: StakeAndSlash<Self::AccountId, BlockNumberFor<Self>, Self::Reward>;
 
-		/// Maximal number of relayers that can register themselves on a single lane.
+		/// Maximal number of relayers that can reside in the active lane relayers set on a single
+		/// lane.
+		///
+		/// Lowering this value leads to additional concurrency between relayers, potentially
+		/// making messages cheaper. So it shall not be too large.
+		#[pallet::constant]
+		type MaxActiveRelayersPerLane: Get<u32>;
+		/// Maximal number of relayers that can reside in the next lane relayers set on a single
+		/// lane.
+		///
+		/// Relayers set is a bounded priority queue, where relayers with lower expected reward are
+		/// prioritized over greedier relayers. At the end of epoch, we select top
+		/// `MaxActiveRelayersPerLane` relayers from the next set and move them to the next set. To
```

Comment:
Please note that it isn't the whole solution - there's yet two additional steps mentioned in https://github.com/paritytech/parity-bridges-common/pull/2605#discussion_r1352615442 and extracted to the #2618